### PR TITLE
Task/cooks 56

### PR DESCRIPTION
**Pull request by taoteg**
## Overview: ##

Initial implementation of the ObservedFeaturesPlot component.
Select a geographic region to populate the chart with data for selected feature.
Current selected feature will be highlighted in chart and key (in orange currently).
Key Column on far right is scrollable with mouse.
Hovering over chart will reveal detailed popup of individual row values.
Click and drag to zoom in further.
Double click to reset chart.
Float values will be rounded to two decimal places and appended with a percent symbol.

## Related Jira tickets: ##

* [COOKS-56](https://jira.tacc.utexas.edu/browse/COOKS-56)

## Summary of Changes: ##

Working implementation of the Observed Features Plot component.

## Testing Steps: ##

1. Build project and run.
2. Select demographic features.
3. Click a region on the map.
4. Select a different Feature to update the plot.

## UI Photos:

Screen Shot 2021-08-17 at 1 10 51 AM

Screen Shot 2021-08-17 at 1 12 22 AM

Screen Shot 2021-08-17 at 1 22 13 AM

### CODE REVIEW

**Comment by nathanfranklin** [comment]
Cool.  Testing now. 👍 

I'll add some minor refactoring (mostly removing unused stuff) and a fix for the scenario when County isn't selected (that is failing at the moment). Will show in pre-prod today to get feed back (Will probably merge 🤔 ).  Thanks for the big push to get this completed 👍

**Review by nathanfranklin — COMMENTED** [review]
> Cool. Testing now. 👍
> 
> I'll add some minor refactoring (mostly removing unused stuff) and a fix for the scenario when County isn't selected (that is failing at the moment). Will show in pre-prod today to get feed back (Will probably merge 🤔 ). Thanks for the big push to get this completed 👍

Looks good.   There is a crash when looking at non-county data and selecting an area that needs to be investigated.

**Comment by nathanfranklin** [comment]
Two other comments that came up with Tracy:

* can we drop the labels on the far right
* can switch the sorting of the counties on the left (from Anderson to Wise instead of Wise to Anderson)

**Comment by taoteg** [comment]
Excellent. Yeah, I can reverse the ordering on the chart.
I am pretty sure I can disable the right side category keys for this chart as well with a unique config object.
I'll update and get that pushed stat for review,

**Comment by taoteg** [comment]
Fixed the crash bug and made the visual adjustments requested. I believe it is now ready to go, though the performance is a bit slow on some of the demographic plots (zip codes in particular is very slow to load, but it does eventually). We can optimize our data object delivery to the front end once we have working prototypes and are in the optimization phase (aka - an actual database of some kind).

Also, testes merging in the new nav refactor and it is zero conflict.

Screen Shot 2021-08-19 at 7 44 50 PM

Screen Shot 2021-08-19 at 7 45 19 PM

Screen Shot 2021-08-19 at 7 45 36 PM

Screen Shot 2021-08-19 at 7 45 55 PM

Screen Shot 2021-08-19 at 7 46 14 PM

This one will *almost* kill your browser...

Screen Shot 2021-08-19 at 7 48 30 PM

File: client/src/components/ProTx/components/util.js [hunk 1]
```diff
@@ -282,48 +317,67 @@ export const getFipsIdName = currentGeoid => {
...
-  const barColor = plotColors[indexKey];
+  if (targetPlot === 'maltreatment') {
+    const indexKey = categoryCodes.indexOf(catcode);
+    barColor = plotColors[indexKey];
+    return barColor;
+  }
+
+  if (targetPlot === 'observed') {
+    if (unique) {
+      barColorIndex = 10;
```
Comment (nathanfranklin) [inline client/src/components/ProTx/components/util.js:368]: never used.

File: client/src/components/ProTx/components/util.js [hunk 2]
```diff
@@ -1,7 +1,40 @@
...
-import { OBSERVED_FEATURES, MALTREATMENT } from './meta';
+import { THEME_CB12_MAIN, THEME_CB12_ALT0 } from './colors';
+import PHR_MSA_COUNTIES from './PHR_MSA_County_Data';
+import { OBSERVED_FEATURES, MALTREATMENT, CATEGORY_CODES } from './meta';
+
+/**
+ * Assign an imported color theme for use in plot generation.
+ */
+export const plotColors = THEME_CB12_MAIN;
+export const histColors = THEME_CB12_ALT0;
```
Comment (nathanfranklin) [inline client/src/components/ProTx/components/util.js:10]: maybe those don't need to be exported?  like in getTraceFillColor, just `THEME_CB12_ALT0` could also be directly used.

File: client/src/components/ProTx/components/util.js [hunk 3]
```diff
@@ -440,13 +541,31 @@ export const getObservedFeaturesLabel = selectedObservedFeatureCode => {
...
+};
+
 /**
  *
  * @param {*} typesDataArray
  * @returns
  */
 export const getPredictiveFeaturesDataObject = () => {
   const newPredictiveFeaturesDataObject = [];
+
```
Comment (nathanfranklin) [inline client/src/components/ProTx/components/util.js:566]: drop

File: client/src/components/ProTx/components/util.js (hunk 2 above)
Comment (taoteg) [reply client/src/components/ProTx/components/util.js:10]: We could certainly drop the exports. There are a number of methods that could use the same cleanup (never used outside utils.js). Lets grab those in a cleanup ticket.

I want to keep the color array variable assignments so I can swap out the plot "themes" in the trace fill method without having to touch the internal plumbing.

File: client/src/components/ProTx/components/util.js (hunk 1 above)
Comment (taoteg) [reply client/src/components/ProTx/components/util.js:368]: Ha! This was bad logic, *should* have been used but wasnt being. It is now. =)

**Pull request merged**